packages/contracts/src/EmailWalletCore.sol

```diff
@@ -513,11 +539,14 @@ contract EmailWalletCore is ReentrancyGuard, OwnableUpgradeable, UUPSUpgradeable
             emailAddrCommit: emailAddrCommit,
             extensionAddr: extensionAddr,
             sender: msg.sender,
-            state: state
+            state: state,
+            expiryTime: expiryTime
         });
 
         Extension extension = Extension(extensionAddr);
-        bool registered = extension.registerUnclaimedState(us);
+        // Sora's comment: Here try-and-catch should be used to prevent a malicious extension that returns error.  
+        // The max gas to call it is calculated from `unclaimedStateRegisterGas`.
```

Comment:
Yes, I found `try-and-catch` is unnecessary here.
I commented on that point to prevent unexpected revert errors in `extension.registerUnclaimedState`, but 1) the error will be caught in `_executeEmailOp` when the running extension registers an unclaimed state internally, and 2) the error should be handled by a caller when it is registered externally.